src/DataFramesMeta.jl

```diff
@@ -295,20 +304,40 @@ end
 ##
 ##############################################################################
 
-where(d::AbstractDataFrame, arg) = d[arg, :]
-where(d::AbstractDataFrame, f::Function) = d[f(d), :]
-where(g::GroupedDataFrame, f::Function) = g[Bool[f(x) for x in g]]
+function where_helper(x, args...)
+    t = (fun_to_vec(arg; nolhs = true, gensym_names = true) for arg in args)
+    quote
+        $where($x, $(t...))
+    end
+end
+
+and(x, y) = x .& y
 
-and(x, y) = :($x .& $y)
+function df_to_bool(res)
+    if any(t -> !(t isa Union{Vector{<:Union{Missing, Bool}}, BitArray{1}}), eachcol(res))
```

Comment:
why `AbstractVector` is not allowed? E.g. from a `SubDataFrame`? Is this tested?

Reply:
Thanks for this. I added a method for `SubDataFrame` for both `@where` and `@orderby` because of this. (needs a new method because `select(subdf, ..., copycols = false)` fails.

Reply:
it is expected to fail if `copycols=false` only column reusing and then `SubDataFrame` is produced (no transformations) are allowed.